packages/bueno/src/values/complex-value.ts

```diff
@@ -54,8 +50,9 @@ export class RecordValue implements SchemaValue<ComplexRecord> {
 
     let out = '';
 
-    for (const [k, v] of Object.entries(input)) {
-      const invalidValue = this.config.values[k].validate(v);
+    for (const [key, validator] of Object.entries(this.config.values)) {
+      const value = input[key];
+      const invalidValue = validator.validate(value);
 
       if (invalidValue !== null) {
         out += ' ' + invalidValue;
```

Comment:
Some notes on the changes in Beuno:

The RecordValue value could not be used with objects that have dynamic keys. The keys of the `f` object are facet ids which are not static.

I removed the check on line 45, and adjusted the loop on 53 to use the configured keys as the ones to validate, versus the keys on the object being validated.

Reply:
Previously, it would check if you passed in an object with "unknown keys", which is different than what we have here right now.

This also explains why you had to modify the corresponding UT (which was checking that it was throwing an error, and now it's not).

I understand that this is needed for facet ids, which we do not know ahead of time.

Checking objects for unknown keys in Bueno might not be extremely useful, mind you... But it could have its use. 

Anyway, I'm fine with the changes you did here. 

I would remove the UT you modified entirely, though, because it's now a useless test.

Reply:
Yeah, I don't know how useful a check for unknown keys in. Also, the logic only compared the number of keys, not the keys themselves.

Regarding the UT, I prefer to keep it because it is still playing a role. It ensures that objected looped on line 53 is the one configured on `RecordValue`, not the input object.

Reply:
Ok, then maybe we can adjust the UT name for it to be more descriptive.